### PR TITLE
Implement Webfinger protocol in setup wizard

### PR DESCRIPTION
**Pull request by fmoc**
Based on #9779. Uses the protocol implementation from https://github.com/owncloud/client/pull/9366. Contributes to #9249.

Opened as a draft because this requires some more cleanup and the error handling could be a little more user friendly.

Tested against the customer's production system.

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/12273/6/1

<details>


</details>

File: src/gui/newwizard/states/webfingersetupwizardstate.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+void WebFingerSetupWizardState::evaluatePage()
+{
+    auto *webFingerSetupWizardPage = qobject_cast<WebFingerSetupWizardPage *>(_page);
+    OC_ASSERT(webFingerSetupWizardPage != nullptr);
+
+    const QString resource = QStringLiteral("acct:%1").arg(webFingerSetupWizardPage->username());
+
+    //    _context->accountBuilder().setWebFingerData(QStringLiteral("acct:bla@example.org"), QUrl(QStringLiteral("https://oauth.example.org")));
```
Comment (TheOneRing) [inline src/gui/newwizard/states/webfingersetupwizardstate.cpp:34]: cleanup?

**Review by TheOneRing — CHANGES_REQUESTED** [review]
Webfinger is supposed to be optional and depending on a branding parameter.

**Review by TheOneRing — CHANGES_REQUESTED** [review]
`07-08 12:55:22:576 [ fatal default ]:	ENSURE: "webFinger->href().isEmpty()" in file /Users/hannah/ownbuild/master/downloads/`

**Comment by fmoc** [comment]
I'll handle that error more gracefully.

**Comment by TheOneRing** [comment]
The user name is the user name and should not contain the webfinger api prefix.
image

**Review by TheOneRing — CHANGES_REQUESTED** [review]
⬆️

**Comment by fmoc** [comment]
You must be using an old state. This has been fixed. https://github.com/owncloud/client/pull/9841/files#diff-fb7667184e5f26f7b04f378cf8d0971deb76e2ffe35f18d138d8eb103494aabbR61

**Comment by TheOneRing** [comment]
> You must be using an old state. This has been fixed. https://github.com/owncloud/client/pull/9841/files#diff-fb7667184e5f26f7b04f378cf8d0971deb76e2ffe35f18d138d8eb103494aabbR61

You are right.

However please set the dav password when we initialize the oauth workflow, this ensures that the id entered by the user is passed to the idp/oauth2 and the user is not asked to enter the id again.

**Pull request merged**